music_this_week_app/templates/music_this_week/setup.html

```diff
@@ -12,23 +12,75 @@
         <p><a href="{{ profile }}" target="_blank">{{ username }}</a>&nbsp; &nbsp;&nbsp;<a href="/login/?retry=true">Switch Users</a></p>
         <p><span style="text-decoration: underline;">Search Arguments:</span></p>
         <form action="/search" id="form">
-            <p><strong>Location:</strong> <input id = "loc_field" type="text" name="location" value="San Francisco"></p>
+            <p id = 'location_options'><strong>Location:</strong> <input id = "loc_field" type="text" name="location" value="{{location}}"> 
+               <input type = "checkbox" id = "latlng_checkbox" name = "latlng_checkbox" onchange = "clickCheckBox.call(this)"> Use my current location </input></p>
             <p><strong>Time range:</strong> <input id = "date_field" type="text" name="date" value="Next 7 Days"></p>
+            <p><input id = "hidden_lat" hidden = true name = "hidden_lat"></p>
+            <p><input id = "hidden_lon" hidden = true name = "hidden_lon"></p>
             <input hidden=true name="nResults" value="200">
             <button type="submit" id="search-button" onclick="buttonClick(this.form)">Search</button>
         </form>
         <p>&nbsp;</p>
         <p id="wait-text" hidden="True">Please be patient. In few minutes a playlist called musicThisWeek will appear in your spotify app.<br><br>If you close this page, the playlist will still be created in the background.</p>
     </div>
-
     <script>
+        //init a variable that will be true if checkbox clicked and user approves location request
+        var userLocationProvided = false;
         //When Search button is pressed, change the UI and then submit the request.
         function buttonClick(form) {
             document.getElementById('search-button').innerHTML = "Searching...";
             document.getElementById('search-button').disabled = "True";
             document.getElementById('wait-text').hidden = false;
             form.submit();
         }
+        //When checkbox is clicked request users location if we havent already
+        function clickCheckBox(){
+            console.log("box checked");
+            console.log(userLocationProvided);
+            var checkbox = document.getElementById('latlng_checkbox');
+            if ((!userLocationProvided)  && (checkbox.checked)){
+                //uncheck box until location obtained, prevents errors from user double click
+                checkbox.checked = false;
+                getUserLocation();
+                }
+            else{
+                toggleLocationField();
+                }            
+
+        }
+        //Hides and Shows location field of form
+        function toggleLocationField(){
+            var loc_field = document.getElementById('loc_field');
+            if (loc_field.hidden){
+                loc_field.hidden = false;
+                }
+            else{
+                loc_field.hidden = true;
+                }
+        }
+
+        function locationProvided(position) {
```

Comment:
Can you please change this function name to something that better indicates 1) what it does and 2) when it would be called? Maybe `getLocationCallback`?